vulnerabilities/views.py

```diff
@@ -19,50 +19,118 @@
                 packages = Paginator(packages, 50).get_page(1)
-            packages = Paginator(packages, 50).get_page(page_no)
+            packages = pages.get_page(page_no)
+
             context["packages"] = packages
             context["searched_for"] = urlencode(
                 {param: request.GET[param] for param in request.GET if param != "page"}
             )
             context["result_size"] = result_size
 
+            if len(request.GET["type"]):
+                package_type = request.GET["type"]
+                context["package_type"] = package_type
+
+            if len(request.GET["name"]):
+                package_name = request.GET["name"]
+                context["package_name"] = package_name
+
+            if result_size == 0:
+                context = {
+                    "package_search": "The VCIO DB does not contain a record of the package you entered -- "
+                    + request.GET["name"]
+                    + ".",
+                    "debug_ui": settings.DEBUG_UI,
+                }
+
+                if request.GET.get("template") == "packages":
+                    context["package_form"] = PackageForm(request.GET or None)
+                    context["template_name"] = "packages.html"
+                    return render(request, "packages.html", context)
+                elif request.GET.get("template") == "package_details":
+                    context["package_form"] = PackageForm(request.GET or None)
+                    context["template_name"] = "package_update.html"
+                    return render(request, "package_update.html", context)
+                elif request.GET.get("template") == "index":
+                    context["package_form"] = PackageForm(request.GET or None)
+                    context["vuln_form"] = CVEForm(request.GET or None)
+                    context["template_name"] = "index.html"
+                    return render(request, "index.html", context)
+                else:
+                    context["package_form"] = PackageForm(request.GET or None)
+                    context["vuln_form"] = CVEForm(request.GET or None)
+                    context["template_name"] = "index.html"
+                    return render(request, "index.html", context)
+
+        context["package_form"] = PackageForm(request.GET or None)
+        context["template_name"] = self.template_name
         return render(request, self.template_name, context)
 
     @staticmethod
     def request_to_queryset(request):
         package_type = ""
         package_name = ""
+        purl = ""
 
         if len(request.GET["type"]):
             package_type = request.GET["type"]
 
         if len(request.GET["name"]):
             package_name = request.GET["name"]
 
+        # Check whether the input value is a syntactically-correct purl
+        try:
+            purl = PackageURL.from_string(package_name)
+            return list(
+                models.Package.objects.all()
```

Comment:
`.all()` is not needed if you call `.filter()` after.

Reply:
See new issue https://github.com/nexB/vulnerablecode/issues/879